pkg/backporter/backporter.go

```diff
@@ -299,44 +339,47 @@ func GetBugHandler(client bugzilla.Client) http.HandlerFunc {
 }
 
 func getClonesTemplateData(bugID int, client bugzilla.Client, allTargetVersions sets.String) (*ClonesTemplateData, int, error) {
-	bug, err := client.GetBug(bugID)
-	if err != nil {
-		return nil, http.StatusNotFound, fmt.Errorf("Bug#%d not found: %w", bugID, err)
-	}
-
-	prs, err := client.GetExternalBugPRsOnBug(bug.ID)
-	if err != nil {
-		return nil, http.StatusInternalServerError, fmt.Errorf("Bug#%d - error occured while retreiving list of PRs: %w", bug.ID, err)
-	}
-
-	parent, err := client.GetRootForClone(bug)
-	if err != nil {
-		return nil, http.StatusInternalServerError, fmt.Errorf("unable to fetch root bug: %w", err)
-	}
-
-	clones, err := client.GetAllClones(bug)
+	clones, bug, root, err := client.GetAllClones(bugID)
 	if err != nil {
 		return nil, http.StatusInternalServerError, fmt.Errorf("unable to get clones: %w", err)
 	}
 	// Target versions would be used to populate the CreateClone dropdown
 	targetVersions := sets.NewString(allTargetVersions.List()...)
 	// Remove target versions of the original bug
 	targetVersions.Delete(bug.TargetRelease...)
-	for _, clone := range clones {
-		clonePRs, err := client.GetExternalBugPRsOnBug(clone.ID)
-		if err != nil {
-			return nil, http.StatusInternalServerError, fmt.Errorf("Bug#%d - error occured while retreiving list of PRs : %w", clone.ID, err)
-		}
-
+	numClones := len(clones)
+	PRErrors := make([]error, numClones+1)
```

Comment:
Looks like `numClones` is only used here, IMO no need for the variable. Also, looks like this could in fact be a channel of errors? Would save you from the need to book-keep the indexes.

Reply:
Yup, removed the numClones variable and switched to the errgroup suggested by Alvaro in another PR